src/Public/OpenApi.ps1

```diff
@@ -1424,11 +2866,22 @@ function Set-PodeOARouteInfo {
     )
 
     foreach ($r in @($Route)) {
-        $r.OpenApi.Summary = $Summary
-        $r.OpenApi.Description = $Description
-        $r.OpenApi.OperationId = $OperationId
-        $r.OpenApi.Tags = $Tags
-        $r.OpenApi.Deprecated = $Deprecated.IsPresent
+        if ($Summary) {
+            $r.OpenApi.Summary = $Summary
+        }
+        if ($Description) {
+            $r.OpenApi.Description = $Description
+        }
+        if ($OperationId) {
+            $r.OpenApi.OperationId = $OperationId
+        }
+        if ($Tags) {
+            $r.OpenApi.Tags = $Tags
+        }
+        $r.OpenApi.Swagger = $true
```

Comment:
I've just realised something about this line, it forces this function to become mandatory on all Routes for them to appear in the OpenAPI definition. I grabbed the original `web-rest-openapi` examples to test them, and all of them still work except for `web-rest-openapi-simple.ps1` because `Add-PodeRoute` isn't being piped into `Set-PodeOARouteInfo`; which makes this a breaking change.

By default all Routes in Pode have some OpenAPI details pre-generated to appear in a basic form within swagger. Is the intent here to make it so that only certain Routes will appear, rather than all by default?

If so, just to keep the current functionality working: the `OpenAPI.Swagger` flag would have to be defaulted to `$true`, and then a new `Hide-PodeOARoute` where piping the Route into this function would set the `OpenAPI.Swagger` flag to `$false`.

Reply:
I see this behavior as bug not a feature. Why should have an swagger entry for something is not openapi?

Reply:
It's documented here as the default behaviour for just using `Add-PodeRoute` on it's own, as a quick way of showing minimal definitions in Swagger for Routes: https://badgerati.github.io/Pode/Tutorials/OpenAPI/#default-setup

One other way could be to add a switch to `Enable-PodeOpenApi` to disable the minimal definitions default (something like `-DisableMinimalDefinitions`):
* When the switch isn't supplied then the `OpenApi.Swagger` property is ignored in `Get-PodeOpenApiDefinitionInternal`, allowing `Set-PodeOARouteInfo` not to be required.
* When the switch is supplied then the `OpenApi.Swagger` property is checked in `Get-PodeOpenApiDefinitionInternal`, and `Set-PodeOARouteInfo` is required to set the property to true.

Reply:
I added a new -DisableMinimalDefinitions option